demos/common/python/openvino/model_zoo/model_api/models/nanodet.py

```diff
@@ -0,0 +1,131 @@
+        self._check_io_number(1, 1)
+        self.output_blob_name = self._get_outputs()
+        self.strides = [8, 16, 32] if self.is_legacy else [8, 16, 32, 64]
+        self.reg_max = 7
+
+    def _get_outputs(self):
+        output_blob_name = next(iter(self.outputs))
+        output_size = self.outputs[output_blob_name].shape
+        if len(output_size) != 3:
+            self.raise_error("Unexpected output blob shape {}. Only 3D output blob is supported".format(output_size))
+
+        return output_blob_name
+
+    @classmethod
+    def parameters(cls):
+        parameters = super().parameters()
+        parameters['resize_type'].update_default_value('fit_to_window')
+        parameters['confidence_threshold'].update_default_value(0.05)
+        parameters.update({
+            'iou_threshold': NumericalValue(default_value=0.6, description="Threshold for NMS filtering"),
+            'num_classes': NumericalValue(default_value=80, description="Number of classes"),
+            'is_legacy': BooleanValue(default_value=False, description='Using a legacy NanoDet model')
+        })
+        return parameters
+
+    def postprocess(self, outputs, meta):
+        detections = self._parse_outputs(outputs, meta)
+        detections = self.rescale_detections(detections, meta)
+        return detections
+
+    def _parse_outputs(self, outputs, meta):
+        output = outputs[self.output_blob_name][0]
+
+        cls_scores = output[:, :self.num_classes]
+        bbox_preds = output[:, self.num_classes:]
+        input_height, input_width = meta['padded_shape'][:2] if meta.get('padded_shape') else meta['resized_shape'][:2]
+
+        bboxes = self.get_bboxes(bbox_preds, input_height, input_width)
+        detections = {'labels': [], 'scores': [], 'x_mins': [], 'y_mins': [], 'x_maxs': [], 'y_maxs': []}
+        for label, score in enumerate(np.transpose(cls_scores)):
+            mask = score > self.confidence_threshold
+            filtered_boxes, score = bboxes[mask, :], score[mask]
+            if score.size == 0:
+                continue
+            x_mins, y_mins, x_maxs, y_maxs = filtered_boxes.T
+            keep = nms(x_mins, y_mins, x_maxs, y_maxs, score, self.iou_threshold, include_boundaries=True)
+            score = score[keep]
+            x_mins, y_mins, x_maxs, y_maxs = x_mins[keep], y_mins[keep], x_maxs[keep], y_maxs[keep]
+            labels = np.full_like(score, label, dtype=int)
+
+            detections['labels'].extend(labels)
+            detections['scores'].extend(score)
+            detections['x_mins'].extend(x_mins)
+            detections['y_mins'].extend(y_mins)
+            detections['x_maxs'].extend(x_maxs)
+            detections['y_maxs'].extend(y_maxs)
+
+        dets = [Detection(*det) for det in zip(detections['x_mins'], detections['y_mins'], detections['x_maxs'],
+                                               detections['y_maxs'], detections['scores'], detections['labels'])]
```

Comment:
Detections might be added to list `dets` without its storage in the dictionary
```suggestion
            dets += [Detection(*det) for det in zip(x_mins, y_mins, x_maxs, y_maxs, score, labels)]
```

Reply:
Done